utils/scm/openebs/delete_cstor_csi_clone.yml

```diff
@@ -0,0 +1,115 @@
+---
+
+## verify that parent pvc should not be deleted
+- name: delete pvc from which snapshot is created 
+  shell: >
+    kubectl delete pvc {{ pvc_name }} -n {{ namespace }}
+  args:
+    executable: /bin/bash
+  register: status
+  failed_when: "status.rc ==0"
+
+## verify snapshot cr should not be deleted
+# - name: delete snapshot cr
+#   shell: >
+#     kubectl delete volumesnapshot.snapshot {{ snapshot_name }} -n {{ namespace }}
+#   args:
+#     executable: /bin/bash
+#   register: status
+#   failed_when: "status.rc ==0"
+
+## store the PV and CVRs for corresponding cloned pvc 
+- name: Derive PV name from PVC
+  shell: >
+    kubectl get pvc {{ cloned_pvc }} -n {{ namespace }}
+    --no-headers -o custom-columns=:spec.volumeName
+  args:
+    executable: /bin/bash
+  register: pv
+
+- set_fact:
+    cloned_pv_name: "{{ pv.stdout }}"
+
+- name: Obtaining the CVR from PV
+  shell: >
+    kubectl get cvr -n {{ operator_ns }} -l cstorvolume.openebs.io/name={{ pv.stdout }} 
+    --no-headers  -o custom-columns=:metadata.name
+  args:
+    executable: /bin/bash
+  register: cvr
+
+- set_fact:
+    cloned_cvr_name: "{{ cvr.stdout }}"
+
+## obtain the application pod name
+- name: Get the app pod name
+  shell: >
+    kubectl get pods -n {{ namespace }} -l app={{ app_label }} 
+    --no-headers -o custom-columns=:.metadata.name
+  args: 
+    executable: /bin/bash
+  register: app_pod_name
+
+- set_fact: 
+    application_podname: "{{ app_pod_name.stdout }}"
+
+## delete the busybox deployment  
+- name: delete the busybox_app 
+  shell: > 
+    kubectl delete deploy {{ app_name }} -n {{ namespace }}
```

Comment:
use 'kubectl delete -f <file>' to delete app deployment

Reply:
resolved